src/renderer/components/NavigationBar/NavigationBar.tsx

```diff
@@ -44,18 +39,16 @@ const NavigationBar = () => {
   const [isExpandLocked, setIsExpandLocked] = useState<boolean>(false);
   const dispatch = useDispatch();
   const currentView = useSelector(selectCurrentView);
+  const activeProject = useSelector(activePublication);
   const { t, i18n } = useTranslation();
 
   const onNavigationButtonClick = (button: IButton) => () => {
     dispatch(updateCurrentView(button.view));
   };
 
   const isButtonActive = (button: IButton) => button.view === currentView.view;
-  const isImageButton = (
-    button: IButton | IImageButton
-  ): button is IImageButton => (button as IImageButton).src !== undefined;
 
-  const renderButton = (button: IButton | IImageButton) => {
+  const renderButton = (button: IButton, isProjectButton?: boolean) => {
```

Comment:
I would usually stay away from boolean parameters. We all get caught up and use them sometimes so good to point it out when possible just so that the muscle memory builds up and we use them less and less. They tend to make the code less readable and require the definition lookup to understand what the boolean is for. This is more problematic in pure JavaScript since in TypeScript it's quite easy to just hover over the function name to see what it could stand for (unless someone codes in a very basic IDE or looks the code up on repo). So in simple cases you can use an options object instead so the param becomes a named one and immediately tells the reader what it stands for -> `renderButton(button, { isProjectButton: true })`. There are couple different solutions but since this is a component only function something like that should do the job just fine.

Here is a good article on the topic of boolean parameters: https://understandlegacycode.com/blog/what-is-wrong-with-boolean-parameters/.

Reply:
Fair point, corrected.